From 142a7c8f128cedc42e09e8d2ed080712681ce48a Mon Sep 17 00:00:00 2001
From: "Unnithan, Balakrishnan" <balakrishnan.unnithan@intel.com>
Date: Wed, 18 Sep 2024 11:47:28 +0530
Subject: [PATCH] fix for OTA update through script failure-2

OTA update was failing due to sepolicy permission denial.
Added sepolicy to open-read-write block-device block- file.

Tests done: Run command ./update_device.py caas-ota-CR0000094.zip

Tracked-On: OAM-123102
Signed-off-by: Unnithan, Balakrishnan <balakrishnan.unnithan@intel.com>
---
 abota/generic/update_engine.te | 2 ++
 virtual_ab/update_engine.te    | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/abota/generic/update_engine.te b/abota/generic/update_engine.te
index bbc7000..20afcb5 100644
--- a/abota/generic/update_engine.te
+++ b/abota/generic/update_engine.te
@@ -15,3 +15,5 @@ allow update_engine mnt_media_rw_file:dir r_dir_perms;
 allow update_engine sdcardfs:dir search;
 allow update_engine sdcardfs:file { read getattr open };
 allow update_engine storage_file:dir search;
+
+allow update_engine block_device:blk_file { open read write };
diff --git a/virtual_ab/update_engine.te b/virtual_ab/update_engine.te
index 1cf0f9b..81c1e1e 100644
--- a/virtual_ab/update_engine.te
+++ b/virtual_ab/update_engine.te
@@ -2,3 +2,5 @@ allow update_engine proc_bootconfig:file r_file_perms;
 allow update_engine vd_device:blk_file rw_file_perms;
 allowxperm update_engine vd_device:blk_file ioctl BLKROGET;
 allow update_engine dm_user_device:chr_file r_file_perms;
+
+allow update_engine block_device:blk_file { open read write };
-- 
2.25.1

